src/main/java/com/example/petbuddybackend/service/care/CareService.java

```diff
@@ -8,177 +8,108 @@
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.domain.Specification;
 import org.springframework.stereotype.Service;
 
 import java.time.LocalDate;
 import java.time.ZoneId;
-import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
 public class CareService {
 
-    @Value("${notification.care.reservation}")
-    private String reservationMessage;
+    private static final String ATTRIBUTE_MISMATCH_FORMAT = "%s (attribute: %s)";
+    private static final String BAD_CARE_RANGE_DATE_MESSAGE = "End care date must be after start care date";
+    private static final String CARETAKER_NOT_OWNER_MESSAGE = "Caretaker is not owner of the care";
+    private static final String CARE = "Care";
+    private static final String CLIENT_NOT_OWNER_MESSAGE = "Client is not owner of the care";
 
-    @Value("${notification.care.update_reservation}")
-    private String updateReservationMessage;
+    private static final String RESERVATION_MESSAGE = "message.care.reservation";
+    private static final String UPDATE_RESERVATION_MESSAGE = "message.care.update_reservation";
+    private static final String ACCEPT_RESERVATION_MESSAGE = "message.care.accepted_reservation";
+    private static final String REJECT_RESERVATION_MESSAGE = "message.care.rejected_reservation";
+    public static final String ANIMAL_ATTRIBUTE_MISMATCH_MESSAGE = "Animal attributes must match animal type. Mismatches: %s";
 
-    @Value("${notification.care.accepted_reservation}")
-    private String acceptReservationMessage;
-
-    @Value("${notification.care.rejected_reservation}")
-    private String rejectReservationMessage;
-    
     private final CareRepository careRepository;
     private final AnimalService animalService;
     private final CaretakerService caretakerService;
     private final ClientService clientService;
+    private final UserService userService;
     private final NotificationService notificationService;
     private final CareMapper careMapper = CareMapper.INSTANCE;
+    private final CareStateMachine careStateMachine;
 
-    public CareDTO makeReservation(CreateCareDTO createCare, String clientEmail, ZoneId timeZone) {
-
-        assertLoggedInUserIsClient(createCare.clientEmail(), clientEmail,
-                "Client can only make reservation for themselves");
-        assertClientIsNotCaretaker(createCare.clientEmail(), createCare.caretakerEmail());
+    public CareDTO makeReservation(CreateCareDTO createCare, String clientEmail, String caretakerEmail, ZoneId timeZone) {
+        userService.assertHasRole(clientEmail, Role.CLIENT);
+        userService.assertHasRole(caretakerEmail, Role.CARETAKER);
         assertEndCareDateIsAfterStartCareDate(createCare.careStart(), createCare.careEnd());
```

Comment:
assertEndCareDateIsAfterStartCareDate jest chyba niepotrzebny bo adnotacja w DTO to sprawdza

Reply:
tak samo w poozstalych metodach

Reply:
Wrzuciłem bo wcześniej było, ale faktycznie moge wyrzucić w sumie. To samo nam `@Check` w końcu załatwia